test/libsolidity/syntaxTests/immutable/uninitialized_private_state_var.sol

```diff
@@ -17,5 +17,7 @@ contract C is B {
 
 }
 // ----
+// TypeError 7733: (196-197): Immutable variables cannot be read before they are initialized.
```

Comment:
Why is this error generated? The compiler should only complain about the variable not being initialized. Maybe we should
1) mark all variables initialized after the construction stage or
2) onyl use `m_initializedStateVar` in the construction stage

Reply:
> The compiler should only complain about the variable not being initialized.

In the case where the variable is not initialized at all we don't want to have read-related errors?

Reply:
Yes, but only for reads at construction time.

Reply:
Are you sure the test case is how you intended it to be?